pkg/common/cluster_state.go

```diff
@@ -171,6 +177,22 @@ func (i *ClusterState) readPostgresqlServiceCurrentState(context context.Context
 	return nil
 }
 
+func (i *ClusterState) readPostgresqlServiceEndpointsCurrentState(context context.Context, cr *kc.Keycloak, controllerClient client.Client) error {
+	postgresqlServiceEndpoints := model.PostgresqlServiceEndpoints(cr)
+	postgresqlServiceEndpointsSelector := model.PostgresqlServiceEndpointsSelector(cr)
+
+	err := controllerClient.Get(context, postgresqlServiceEndpointsSelector, postgresqlServiceEndpoints)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			return err
+		}
+	} else {
+		i.PostgresqlServiceEndpoints = postgresqlServiceEndpoints.DeepCopy()
+		cr.UpdateStatusSecondaryResources(i.PostgresqlService.Kind, i.PostgresqlService.Name)
```

Comment:
```suggestion
		cr.UpdateStatusSecondaryResources(i.PostgresqlServiceEndpoints.Kind, i.PostgresqlServiceEndpoints.Name)
```

Reply:
This is unfortunately also a problem as this Endpoints resource type will get added to the secondary resources in the status even when an external database is not added.  This is also a problem for the in the case where externalAccess is false for a route or an ingress. We should probably not be reading their state from the cluster in cases where we know we won't need them.

Reply:
Good point. I'll remove it from the secondary resources.

Reply:
Actually, I have a better idea! I'll add the Endpoints only if the external database support is turned on.

Reply:
perfect now, thank you